homeassistant/components/homekit/__init__.py

```diff
@@ -430,11 +433,18 @@ def setup(self):
             advertised_address=self._advertise_ip,
             interface_choice=self._interface_choice,
         )
+
         self.bridge = HomeBridge(self.hass, self.driver, self._name)
         if self._safe_mode:
             _LOGGER.debug("Safe_mode selected for %s", self._name)
             self.driver.safe_mode = True
 
+    async def async_setup_zeroconf(self):
+        """Share the system zeroconf instance."""
+        # Replace the existing zeroconf instance.
+        await self.hass.async_add_executor_job(self.driver.advertiser.close)
+        self.driver.advertiser = await zeroconf.async_get_instance(self.hass)
```

Comment:
This is fine for now but it feels a little weird. What about any listeners that have been set on the advertiser? 

Could we consider updating the driver to allow passing in an advertiser instance ?

Reply:
At this point it is an empty so no concerns about loosing any listeners.

I'm going send an upstream PR to allow passing in the instance after the other round of fixes I have in `HAP-python` gets released.

Reply:
`HAP-python` was updated, and this is now cleaned up here https://github.com/home-assistant/core/pull/35687